File: Lista de Exercícios/Go/11. Classes Abstratas/ClassesAbstratas.go
package main

import (
	"fmt"
)

type Funcionario interface {
	calcularSalario() float64
}

type FuncionarioHorista struct {
	valorHora       float64
	horasTrabalhadas float64
}

func (f FuncionarioHorista) calcularSalario() float64 {
	return f.valorHora * f.horasTrabalhadas
}

type FuncionarioAssalariado struct {
	salario float64
}

func (f FuncionarioAssalariado) calcularSalario() float64 {
	return f.salario
}

func main() {

}

File: Lista de Exercícios/Java/11. Classes Abstratas/ClassesAbstratas.java
abstract class Funcionario {
    public abstract double calcularSalario();
}

class FuncionarioHorista extends Funcionario {
    private double valorHora;
    private int horasTrabalhadas;

    public FuncionarioHorista(double valorHora, int horasTrabalhadas) {
        this.valorHora = valorHora;
        this.horasTrabalhadas = horasTrabalhadas;
    }

    @Override
    public double calcularSalario() {
        return valorHora * horasTrabalhadas;
    }
}

class FuncionarioAssalariado extends Funcionario {
    private double salario;

    // Construtor
    public FuncionarioAssalariado(double salario) {
        this.salario = salario;
    }

    @Override
    public double calcularSalario() {
        return salario;
    }
}